src/lua_channel.cpp

```diff
@@ -844,8 +849,13 @@ int LuaChannel::removeMod(lua_State* L) {
     lua_pop(L, 2);
 
     chan->remMod(dest);
-    if (chan->getType() == CT_PUBLIC)
+    if (chan->getType() == CT_PUBLIC) {
         ServerState::rebuildChannelOpList();
+        auto con = ServerState::getConnection(dest);
+        if(con && !ServerState::isChannelOp(dest)) {
```

Comment:
I may be missing something here; why the second condition?

Reply:
They should only have the role removed if they are not a channel op in another public channel.